plugins/source_testing.go

```diff
@@ -21,13 +20,15 @@ type ResourceTestCase struct {
 	SkipIgnoreInTest bool
 }
 
-func init() {
-	_ = faker.SetRandomMapAndSliceMinSize(1)
-	_ = faker.SetRandomMapAndSliceMaxSize(1)
+type TestSourcePluginOptions struct {
+	NoParallel bool
 }
 
-func TestSourcePluginSync(t *testing.T, plugin *SourcePlugin, logger zerolog.Logger, spec specs.Source) {
+func TestSourcePluginSync(t *testing.T, plugin *SourcePlugin, logger zerolog.Logger, spec specs.Source, opts TestSourcePluginOptions) {
 	t.Helper()
+	if !opts.NoParallel {
```

Comment:
This can be removed for now, but forcing plugin authors to mark their every-syncing-test to be parallel doesn't sound good.

Reply:
I think default sounds good but let's make it more goish with `opts ...TestSourcePluginOption` and introduce `With`

Reply:
logger probably also can go into option